RadixWallet/Features/ChooseAccounts/ChooseAccounts+Reducer.swift

```diff
@@ -112,17 +116,41 @@ struct ChooseAccounts: Sendable, FeatureReducer {
 	func reduce(into state: inout State, presentedAction: Destination.Action) -> Effect<Action> {
 		switch presentedAction {
 		case .createAccount(.delegate(.completed)):
-			loadAccounts()
+			loadAccounts(context: state.context)
 
 		default:
 			.none
 		}
 	}
 
-	private func loadAccounts() -> Effect<Action> {
+	private func loadAccounts(context: State.Context) -> Effect<Action> {
 		.run { send in
 			let result = await TaskResult {
-				try await accountsClient.getAccountsOnCurrentNetwork()
+				switch context {
+				case .assetTransfer, .permission:
+					return try await accountsClient.getAccountsOnCurrentNetwork()
+						.map { State.AccountType.general($0) }
+						.asIdentified()
+
+				case .accountDeletion:
+					let accounts = try await accountsClient.getAccountsOnCurrentNetwork()
+					let entities = try await onLedgerEntitiesClient.getAccounts(accounts.map(\.address), cachingStrategy: .forceUpdate)
+					let receivingAccounts = accounts.compactMap { account -> State.ReceivingAccountCandidate? in
+						guard let entity = entities.first(where: { $0.address == account.address }) else {
+							assertionFailure("Failed to find account, this should never happen.")
+							return nil
+						}
+
+						let xrdBalance = entity.fungibleResources.xrdResource?.amount.exactAmount?.nominalAmount ?? 0
+						let hasEnoughXRD = xrdBalance >= 1
+
+						return .init(account: account, hasEnoughXRD: hasEnoughXRD)
+					}
+					return receivingAccounts
+						.sorted { $0.hasEnoughXRD && !$1.hasEnoughXRD }
+						.map { State.AccountType.receiving($0) }
+						.asIdentified()
+				}
```

Comment:
This logic is too specific for the account delete usecase, I think it should be in ChooseReceivingAccountOnDelete and not here. To facilitate this I would allow injecting the accounts loader in ChoseAccounts+Reducer, that would allow having custom loading logic depending on the context where it is used.